### ISSUE TITLE
matplotlib.use('Agg') conflicts with user code

### ISSUE THREAD

**Issue by LukasDrude** [chainer/chainer#2146]
In this line in `chainer/chainer/training/extensions/plot_report.py` the `matplotlib` backend is changed [Source](https://github.com/pfnet/chainer/blob/master/chainer/training/extensions/plot_report.py#L16):

    matplotlib.use('Agg')

Unfortunately, this can interfere with users code. For example, when the user sets the backend himself anywhere, it is not known, whether his setting or the Chainer settings wins (is imported first).

The `plot_report` gets imported, when `extensions` is imported. For now, I just removed from the corresponding `__init__` file locally, which is definitely not a clean solution.

**Comment by wkentaro** [comment]
I have same problem, and running `matplotlib.use('XXX')` in module seems not good idea.

**Comment by delta2323** [comment]
Hi, thank you for reporting the problem. I agree with removing `matplotlib.use` from PlotReport. But I found another problem. Once `matplotlib.pyplot` is imported, as we do in PlotReport, any call of `matplotlib.use` does not take effect. So import of `PlotReport` can disable user's call of `use()` function. See the [document of matplotlib](http://matplotlib.org/faq/usage_faq.html#what-is-a-backend) for detail.

**Comment by rezoo** [comment]
Hi, it seems that the current branch does not solve an issue raised by @delta2323 .

For example, assume that someone want to import the external modules in alphabetical order like that:
```python
from chainer.training import extensions
import matplotlib
matplotlib.use('Agg')
```
But it raises a UserWarning and takes no effect. In order to enable Agg, we should import modules like
```python
import matplotlib
matplotlib.use('Agg')
from chainer.training import extensions
```

_This specification is not very desirable._ I think we have to reimplement `PlotReport` that does not affect the order of modules.

**Comment by delta2323** [comment]
I agree that import of modules in alphabetical order should not issue warnings because it is recommended by PEP8.

**Comment by beam2d** [comment]
It should've been solved via #2740.

**Comment by eduardomoraes** [comment]
A question for you guys: if a use matplotlib.use('Agg'), how can I disable this line of code and make matplotlib generates plots again?? 
I want to do it without having to restart the notebook, Thanks
